### PR TITLE
Adding anonymous id and aliasing for Outbound.io

### PR DESCRIPTION
**Pull request by usmanity**
This pull request adds the ability to alias using Outbound's client-side implementation and the ability to pass in anonymous users into the identify call.

### CODE REVIEW

**Comment by amillet89** [comment]
@usmanity can u add some tests to cover these new changes? thanks!

**Comment by usmanity** [comment]
@amillet89 tests have been added and updated for these changes.

**Comment by f2prateek** [comment]
@usmanity Could you please squash and rebase these commits into one? Only ` lib/outbound/test.js` and ` lib/outbound/index.js` files should be changed for this commit.

File: lib/outbound/test.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ describe('Outbound', function(){
   var outbound;
   var analytics;
   var options = {
-    publicApiKey: 'pub-3e2b0899b2c81c6f0c59342d1ff057c3'
+    publicApiKey: 'pub-9cb1d6e54b003d5274e54a483ef741be'
```
Comment (f2prateek) [inline lib/outbound/test.js]: We should leave this as the old key right @amillet89? I'm guessing this one is @usmanity's personal Outbound.io key?

**Comment by usmanity** [comment]
@f2prateek squashed and updated. :smile:

**Comment by f2prateek** [comment]
Thanks, LGTM. Just one last nit about the key.

File: lib/outbound/test.js (hunk 1 above)
Comment (usmanity) [inline lib/outbound/test.js]: I've added this key to avoid Invalid API Key error. It is a development key.

**Comment by usmanity** [comment]
@f2prateek and @amillet89, thanks for taking care of this!

**Pull request merged**